internal/scm/github/github.go

```diff
@@ -221,23 +228,24 @@ func (g *Github) getRepositories(ctx context.Context) ([]*github.Repository, err
 		}
 		allRepos = append(allRepos, repos...)
 	}
-
-	// Remove duplicate repos
-	repoMap := map[string]*github.Repository{}
-	for _, repo := range allRepos {
-		repoMap[repo.GetFullName()] = repo
-	}
-	allRepos = make([]*github.Repository, 0, len(repoMap))
-	for _, repo := range repoMap {
-		if repo.GetArchived() || repo.GetDisabled() {
-			continue
-		}
-		allRepos = append(allRepos, repo)
-	}
+	filteredRepos := make([]*github.Repository, 0, len(allRepos))
+	// Filter repositories
+	filteredRepos = slices.DeleteFunc(allRepos, func(repo *github.Repository) bool {
+		regExMatch := repoRegEx != nil && repoRegEx.MatchString(repo.GetFullName())
+		return (!regExMatch && repoRegEx != nil) || (repo.GetArchived() || repo.GetDisabled())
+	})
+	// Remove duplicates
+	allRepos = make([]*github.Repository, 0, len(filteredRepos))
+	slices.SortFunc(filteredRepos, func(g2, g *github.Repository) int {
+		return cmp.Compare(g2.GetFullName(), g.GetFullName())
+	})
+	allRepos = slices.CompactFunc(filteredRepos, func(g2 *github.Repository, g *github.Repository) bool {
+		return g2.GetFullName() == g.GetFullName()
+	})
+	// Sort by Datetime
 	sort.Slice(allRepos, func(i, j int) bool {
 		return allRepos[i].GetCreatedAt().Before(allRepos[j].GetCreatedAt().Time)
 	})
```

Comment:
Since `slices.SortFunc` is used above, should the same be done here?
```suggestion
	slices.SortFunc(filteredRepos, func(g2, g *github.Repository) int {
		return g2.GetCreatedAt().Time.Compare(g.GetCreatedAt().Time)
	})
```